code/modules/food_and_drink/cakes.dm

```diff
@@ -24,62 +28,508 @@
 /obj/item/reagent_containers/food/snacks/cake/custom
 	name = "cake"
 	desc = "a cake"
-	icon_state = "cake"
+	icon_state = "cake_batter"
 	amount = 10
 	heal_amt = 2
 	use_bite_mask = 0
-	var/icing = 0
-	var/sliced = 0
-	var/icing_color =  "#FFFFFF"
 	flags = FPRINT | TABLEPASS | NOSPLASH
+	object_flags = IGNORE_CONTEXT_CLICK_HELD
 	initial_volume = 100
+	w_class = 4.0
+	var/sliced = 0
+	var/static/list/frostingstyles = list("classic","top swirls","bottom swirls","spirals","rose spirals")
+	var/clayer = 1
+	var/amount2 //holds the amount of slices in cake 2 (not entirely, but its used for a bit of math later)
+	var/amount3 //same for cake 3
+	var/cake_candle
+	var/litfam //is the cake lit (candle)
+
+	New()
+		..()
+		contextLayout = new /datum/contextLayout/default()
+		for(var/datum/contextAction/C in src.contextActions)
+			C.dispose()
+		src.contextActions = list()
+
+	/*_______*/
+	/*Utility*/
+	/*‾‾‾‾‾‾‾*/
+	proc/check_for_topping(var/obj/item/W)
+		var/tag = 0
+		var/pendinglight = 0
+		if(istype(W,/obj/item/device/light/candle)) //special handling for candles because they need to send unique information
+			var/obj/item/device/light/candle/candle = W
+			if(candle.on)
+				pendinglight = 1
+				tag = "cake[clayer]-candle_lit"
+			else
+				tag = "cake[clayer]-candle"
+			cake_candle = tag
+		else
+			switch(W.type)
```

Comment:
no consideration for `else`?

Reply:
I'm not entirely sure what the issue is here...

Reply:
What if W is not one of those types? you will return a `list(0,0`

Reply:
fixed fixed! i added a check after the switch statement to make sure there's actually a tag being applied

Reply:
ok so adding the check after the switch statement sort of derailed cake slicing. there's a check on line 494 that checks if the list is (0,0) and performs the default attackby proc.